contracts/REXLimitOrderManager.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity ^0.8.0;
+
+import {ISuperfluid} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+
+import {ISuperToken} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol";
+
+import {SuperTokenV1Library} from "@superfluid-finance/ethereum-contracts/contracts/apps/SuperTokenV1Library.sol";
+
+import "./gelato/AutomateTaskCreator.sol";
+
+import "./IREXUniswapV3Market.sol";
+
+/**
+ * @title Ricochet's Limit Order Manager
+ *
+ * @notice This contract manages limit orders in the Ricochet protocol.
+ *
+ */
+contract REXLimitOrderManager is AutomateTaskCreator {
+    using SuperTokenV1Library for ISuperToken;
+
+    struct LimitOrder {
+        bool isInverted;
+        int96 streamRate;
+        uint256 price;
+        bytes32 taskId;
+        bool executed;
+    }
+
+    constructor(
+        address _automate,
+        address _fundsOwner
+    ) AutomateTaskCreator(_automate, _fundsOwner) {}
+
+    /**
+     * @notice This mapping stores the limit orders for each user.
+     * @dev The mapping is indexed by user address and market address.
+     *
+     */
+    mapping(address => mapping(address => LimitOrder)) public limitOrders;
+
+    event LimitOrderCreated(
+        address _user,
+        address _market,
+        bool _isInverted,
+        int96 _streamRate,
+        uint256 _price
+    );
+
+    /**
+     *
+     * @notice Creates a new limit order.
+     * @dev The order is created with the specified parameters.
+     * @param _market The address of the REX market where the order is placed for.
+     * @param _isInverted A boolean indicating if the order is for an inverted market.
+     * @param _streamRate The streaming rate for the order.
+     * @param _price The price at which the order is placed.
+     */
+    function createLimitOrder(
+        address _market,
+        bool _isInverted,
+        int96 _streamRate,
+        uint256 _price
+    ) external {
+        IREXUniswapV3Market market = IREXUniswapV3Market(_market);
+        ISuperToken token = ISuperToken(market.inputToken());
+
+        require(
+            token.isOperatorFor(address(this), msg.sender),
+            "No ACL permission"
```

Comment:
Use a 2-3 character code for the message

Reply:
Done 👌